cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java

```diff
@@ -57,51 +84,64 @@ private static Map<String, Map<String, Double>> loadStoragePriceList() {
     }
 
     @Cacheable(cacheNames = "awsCostCache", key = "{ #region, #instanceType }")
-    public double getPriceForInstanceType(String region, String instanceType) {
-        PriceListElement priceListElement = getPriceList(region, instanceType);
-        return priceListElement.getTerms().getOnDemand().values().stream().findFirst().orElseThrow(() ->
-                        new NotFoundException(String.format("Couldn't find the price for region [%s] and instance type [%s].", region, instanceType)))
-                .getPriceDimensions().values().stream().findFirst().orElseThrow(() ->
-                        new NotFoundException(String.format("Couldn't find the price for region [%s] and instance type [%s].", region, instanceType)))
-                .getPricePerUnit().getUsd();
+    public Optional<Double> getPriceForInstanceType(String region, String instanceType, ExtendedCloudCredential extendedCloudCredential) {
```

Comment:
if we are caching based on only the region and instanceType wouldn't that mean that we store the first values whichever account gets the first request, and serve the price based on that?
That might be okay in the short term since this will fix pricing problems for customers who don't have policies for getProducts calls.